src/deriveAddress.c

```diff
@@ -83,77 +163,23 @@ static void deriveAddress_return_ui_runStep()
 	UI_STEP_BEGIN(ctx->ui_step, this_fn);
 
 	UI_STEP(RETURN_UI_STEP_WARNING) {
-		#ifdef HAVE_BAGL
-		ui_displayPaginatedText(
-		        "Unusual request",
-		        "Proceed with care",
-		        this_fn
-		);
-		#elif defined(HAVE_NBGL)
-		set_light_confirmation(true);
-		display_warning(
-		        "Unusual request",
-		        this_fn,
-		        respond_with_user_reject
-		);
-		#endif // HAVE_BAGL
+		_displayUnusualRequestWarning(this_fn);
```

Comment:
The ui state machines are hard to read with all those `#ifdef` branches (our UI_STEP macros are complicated enough on their own), so I prefer to extract the platform-specific details into a function with a descriptive name.
@sgliner-ledger what do you think?

Reply:
Very good idea, thanks for these changes !